detach/detach.go

```diff
@@ -0,0 +1,97 @@
+package detach
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+// Hooks are functions that are called with specific detach helpers to attach values from the parent context to the new
```

Comment:
🔕 if you're not going to wrap based on full sentences, 80col is where I would recommend wrapping; this overflows my github review width